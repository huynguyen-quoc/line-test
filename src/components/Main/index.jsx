import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

class Main extends React.PureComponent {
  static propTypes = {
    children: PropTypes.any.isRequired,
  };

  render() {
    const { children } = this.props;
    return (
      <main className="flex-shrink-0">
        <div className="container">{children}</div>
      </main>
    );
  }
}

export default Main;
