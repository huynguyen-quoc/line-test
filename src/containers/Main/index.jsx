import React from 'react';
import { Main as MainComponent, LTable, Pagination } from 'components';
import { generatePages } from 'utils/helpers';
import { Data } from '../../constants';

const DEFAULT_STATE = {
  items: Data,
  pages: generatePages(Data.length / 10),
  pageIndex: 0,
  pageSize: 10,
};

export const DataContext = React.createContext(DEFAULT_STATE);

class Main extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = DEFAULT_STATE;
    this.pageChanged = this.pageChanged.bind(this);
  }

  pageChanged(pageIndex) {
    this.setState({ pageIndex });
  }

  render() {
    return (
      <MainComponent>
        <DataContext.Provider
          value={{
            ...this.state,
          }}
        >
          <DataContext.Consumer>
            {({ items, pageIndex, pageSize, pages }) => (
              <React.Fragment>
                <LTable items={items} pageIndex={pageIndex} pageSize={pageSize} />
                <Pagination
                  items={items}
                  pageIndex={pageIndex}
                  pages={pages}
                  pageSize={pageSize}
                  pageChanged={this.pageChanged}
                />
              </React.Fragment>
            )}
          </DataContext.Consumer>
        </DataContext.Provider>
      </MainComponent>
    );
  }
}
export default Main;
