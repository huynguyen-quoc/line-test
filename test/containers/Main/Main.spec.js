import React from 'react';
import Main from 'containers/Main';

describe('Main', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Main>
        <div>test</div>
      </Main>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
