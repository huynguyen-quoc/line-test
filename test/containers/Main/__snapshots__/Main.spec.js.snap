// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
<Main>
  <ContextProvider
    value={
      Object {
        "alert": Object {
          "isShowAlert": false,
          "message": "",
          "type": "info",
        },
        "detail": Object {
          "cargo_type": Array [],
          "description": "",
          "dimension": "",
          "driver": Array [],
          "parking_address": "",
          "price": "",
          "production_in_year": "",
          "status": "",
          "truck_plate": "",
          "truck_type": "",
        },
        "errors": Object {},
        "isShow": false,
        "items": Array [],
        "modalTitle": "Create new truck",
        "pageIndex": 0,
        "pageSize": 10,
        "pages": Array [],
        "sortField": "",
      }
    }
  >
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </ContextProvider>
</Main>
`;
